From f4e3ac06364ddba7ff8d76c9338f7cb3dae7a952 Mon Sep 17 00:00:00 2001
From: "Hayataka@kitakar5525" <34676735+kitakar5525@users.noreply.github.com>
Date: Tue, 2 Apr 2019 16:37:00 +0900
Subject: [PATCH 1/4] Comment out Error lines and increment revision

---
 dsdt.dsl | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/dsdt.dsl b/dsdt.dsl
index c214d37..50676c8 100644
--- a/dsdt.dsl
+++ b/dsdt.dsl
@@ -1,7 +1,7 @@
-Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload-496)
-Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload2-477)
-Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload-496)
-Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload2-477)
+//Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload-496)
+//Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload2-477)
+//Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload-496)
+//Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (20190215/dswload2-477)
 /*
  * Intel ACPI Component Architecture
  * AML/ASL+ Disassembler version 20190215 (64-bit version)
@@ -22,7 +22,7 @@ Firmware Error (ACPI): Could not resolve symbol [^GFX0.CLID], AE_NOT_FOUND (2019
  *     Compiler ID      "INTL"
  *     Compiler Version 0x20160422 (538313762)
  */
-DefinitionBlock ("", "DSDT", 2, "MSFT  ", "MSFT    ", 0x00000000)
+DefinitionBlock ("", "DSDT", 2, "MSFT  ", "MSFT    ", 0x00000001)
 {
     External (_GPE.AL6F, MethodObj)    // 0 Arguments
     External (_GPE.HLVT, MethodObj)    // 0 Arguments
-- 
2.21.0

